apps/condo/domains/common/utils/pdf.ts

```diff
@@ -1,33 +1,71 @@
 import html2canvas from 'html2canvas'
 import Jspdf from 'jspdf'
 
+const PDF_FORMAT_SETTINGS = {
+    // pdfWidth - width of final image (in mm)
+    // pdfHeight - height of final image (in mm)
+    // elementOffset - offset of image (in mm)
+    // firstLineOffset - margin right for first line (in css pixels)
+    // lineSpace - margin right for rest of lines (in css pixels)
+    'a4': { pdfWidth: 210, pdfHeight: 297, elementOffset: 10, firstLineOffset: 23, lineSpace: 80, ratio: Math.sqrt(2) },
+    'a5': { pdfWidth: 148, pdfHeight: 210, elementOffset: 10, firstLineOffset: 23, lineSpace: 80, ratio: Math.sqrt(2) },
+}
+
 function getPdfHeightFromElement (element: HTMLElement, expectedWidth: number) {
     const { clientWidth, clientHeight } = element
     const originalRatio = clientHeight / clientWidth
     return expectedWidth * originalRatio
 }
 
+function getLine () {
+    const line = document.createElement('div')
+    // line.style.width = 'calc(100% + 48px)'
+    line.style.width = '100%'
+    // line.style.margin = '0 -24px'
+    line.style.borderTop = '1px solid black'
+    return line
+}
+
 interface ICreatePdfOptions {
     element: HTMLElement
     fileName: string
+    format: string
 }
 
 export function createPdf (options: ICreatePdfOptions) {
```

Comment:
I think it would be good to have a great explanation here on what goes where. Code with a lot of math equations might be hard to understand without a docstring

Reply:
> I think it would be good to have a great explanation here on what goes where. Code with a lot of math equations might be hard to understand without a docstring

added few explanation comments in the most difficult parts